components/CredentialBase.vue

```diff
@@ -0,0 +1,157 @@
+<template>
+  <div class="column justify-evenly">
+    <div
+      :class="dense ? 'q-py-none' : 'q-py-xs'"
+      class="cb-content col row justify-between items-center">
+      <slot name="description">
+        <credential-field
+          class="col"
+          :title="nameOverride.length > 0 ? nameOverride : credentialName"
+          title-class="text-left text-subtitle1"
+          :value="descriptionOverride.length > 0 ?
+            descriptionOverride : description"
+          value-class="text-left text-body2 text-grey-7" />
+      </slot>
+      <slot name="image">
+        <dynamic-image
+          :src="imageOverride.length > 0 ? imageOverride : credentialImage"
+          :size="dense ? 'sm' : 'md'" />
+      </slot>
+    </div>
+    <q-separator
+      v-if="(clickable || expandable) && separator"
+      class="cb-separator" />
+    <div
+      :class="dense ? 'q-py-none' : 'q-py-xs'"
+      class="col">
+      <div
+        v-if="clickable"
+        class="cb-clickable cursor-pointer
+        row items-center justify-center text-grey-7"
+        @click="toggleDetails">
+        <span class="cb-details-text q-mr-sm">{{detailsText}}</span>
+        <q-icon :name="detailsIcon" />
+        <q-dialog v-model="state.details">
+          <slot name="modal">
+            <q-card>
+              <slot name="modalHeader" />
+              <slot name="modalContent">
+                <credential-switch
+                  v-bind="$attrs"
+                  mode="details"
+                  :credential="credential" />
+              </slot>
+              <slot name="modalFooter" />
+            </q-card>
+          </slot>
+        </q-dialog>
+      </div>
+      <div
+        v-else-if="expandable"
+        class="cb-expandable row justify-start text-grey-7 full-width">
+        <q-expansion-item
+          :label="detailsText"
+          class="full-width"
+          dense
+          dense-toggle
+          switch-toggle-side>
+          <slot name="expansion">
+            <slot name="expansionHeader" />
+            <slot name="expansionContent">
+              <credential-switch
+                v-bind="$attrs"
+                mode="details"
+                :credential="credential" />
+            </slot>
+            <slot name="expansionFooter" />
+          </slot>
+        </q-expansion-item>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup>
+/*!
+ * Copyright (c) 2018-2022 Digital Bazaar, Inc. All rights reserved.
+ */
+import DynamicImage from './DynamicImage.vue';
+import CredentialField from './CredentialField.vue';
+import CredentialSwitch from './CredentialSwitch.vue';
+import {computed, defineProps, reactive, toRef, unref} from 'vue';
+import {useCredentialCommon} from './credentialCommon.js';
```

Comment:
These should be sorted ... @davidlehn can you help with the eslint rule here?

Reply:
Sorted here: 4a1122145b9b4b1c9023ab51062930e14c2fe365